### PR TITLE
hponcfg - revamped the module using ModuleHelper

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Revamped the module using ModuleHelper. By default that adds the improvement on calling `run_command` with a list instead of a string.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/modules/remote_management/hpilo/hponcfg.py

### CODE REVIEW

File: plugins/modules/remote_management/hpilo/hponcfg.py [hunk 1]
```diff
@@ -69,43 +69,38 @@
...
 
-    if rc != 0:
-        module.fail_json(rc=rc, msg="Failed to run hponcfg", stdout=stdout, stderr=stderr)
+    def __run__(self):
+        self.run_command(params=['src', 'verbose', 'minfw'])
 
-    module.exit_json(changed=changed, stdout=stdout, stderr=stderr)
 
+def main():
+    HPOnCfg.execute()
```
Comment (russoz) [inline plugins/modules/remote_management/hpilo/hponcfg.py:103]:
```suggestion
    HPOnCfg.execute()
    
```

**Review by felixfontein — APPROVED** [review]
Looks good to me.

**Comment by felixfontein** [comment]
Will merge in ~a week if nobody complains.

**Comment by felixfontein** [comment]
@russoz thanks for this one as well!

**Pull request merged**